usr/lib/tik/modules/pre/10-welcome

```diff
@@ -3,34 +3,43 @@
 # SPDX-FileCopyrightText: Copyright 2024 Richard Brown
 # SPDX-FileCopyrightText: Copyright 2024 Raymond Yip
 
+displayWarning=false
+
 proceedInstall() {
 	d --info --ok-label="Install Now" --no-wrap --width=300 --height=300 --icon=distributor-logo-Aeon-symbolic --title="" --text="<big>Welcome to ${TIK_OS_NAME}</big>\n\nPlease press <b>Install Now</b> to continue"
 }
 
 warningMSG() {
-	d --warning --no-wrap --title="AC Power Required" --text="Please connect the AC power adapter to your machine to continue" 
+	d --warning --no-wrap --title="AC Power Recommended" --text="Runnning on battery power detected\n\nIt is recommended to connect the system to AC power during the install"
 }
 
 checkLaptop() {
-	chassis=`cat /sys/class/dmi/id/chassis_type`
-	#Test for respectively Handheld, Notebook, Laptop, and Portable
-	if [ $chassis = "11" ] || [ $chassis = "10" ] || [ $chassis = "9" ] || [ $chassis = "8" ]; then
-		#Tested machine is confirmed mobile
-		#Check for AC power connection
-		#Check file exists first
-		if [ -f "/sys/class/power_supply/AC/online" ]; then
-			#File exists
-			#While the AC power cable is still disconnected
-			while (grep -Fxq "0" "/sys/class/power_supply/AC/online"); do
-				log "AC power disconnected"
-				#Display warning message
-				warningMSG
-			done
-		fi
-	fi
+        chassis=`cat /sys/class/dmi/id/chassis_type`
+        #Test for respectively Handheld, Notebook, Laptop, and Portable
+        if [ $chassis = "11" ] || [ $chassis = "10" ] || [ $chassis = "9" ] || [ $chassis = "8" ]; then
```

Comment:
In this case, you can reverse the if check and do an early return. This way, you eliminate one level of indentation for the actual body of the function, makes it more readable.

Reply:
Please elaborate what you mean here. I was planning to replace this with the following for readability:
`if [[ "$chassis" =~ ^(8|9|10|11)$ ]]; then`

Reply:
Sure :) , and maybe I missed something, but what I meant was roughly (I don't know the correct bash syntax by heart):

```
checkLaptop() {
if [ $chassis != "11" ] && [ $chassis != "10" ] && [ $chassis != "9" ] && [ $chassis != "8" ]; then
# do nothing, in other languages you'd write a return 'return' statement, this is what I meant by early return
fi

# ... do what the function needs to do, but everything is now at the top level instead nested inside an if statement

Reply:
Maybe
[[ "$chassis" =~ ^(8|9|10|11)$ ]] || return

Reply:
Thank you, I made those changes and committed them

Reply:
thank you!
my explanation was a bit misleading, was before my first coffiee!